src/lib/components/Layers/Layers.js

```diff
@@ -18,10 +19,33 @@ export class Layers extends Component {
     this.state = groups;
   }
 
+  componentDidUpdate() {
+    if (this.props.layers) {
+      /** Check if updated children down the hierarchy have
+       * layers which are a visible. If so open the group
+       */
+      this.props.layers.forEach(layer => {
+        if (!layer.id) {
+          Object.keys(layer).forEach(groupName => {
+            const children = layer[groupName].layers;
+
+            if (
+              getMenuGroupVisibleLayers(groupName, children).length &&
+              !this.state[groupName].isOpen
+            ) {
+              this.setState({
+                [groupName]: { isOpen: true },
+              });
+            }
+          });
+        }
+      });
+    }
+  }
+
```

Comment:
@ciremusyoka it feels like every time the code reaches line 36 and sets the state, this lifecycle method will be triggered again and start looping through the layers again. Might it be possible to:

1. Set the state only once after completing the forEach loop?
2. Add something to state which enables/disables this process of looping through the layers?

Reply:
@cKellyDesign this part are some of the parts we talked about being implemented on  [map state to url](https://github.com/onaio/gisida-react/pull/195).
@kahummer @kelvin-muchiri can respond to this.

Reply:
@cKellyDesign after some testing i found we could achieve this without using `componentDidUpdate` and i did away with it.